### ISSUE TITLE
Existing hosting is unable to build `v15|16` dependencies

### ISSUE THREAD

**Issue by jjperezaguinaga** [hoprnet/myne-chat#81]
## Expected Behavior

The application can be built and hosted without any issues.

## Current Behavior

[Vercel](https://vercel.com), our current hosting provider, only has support for node.js instances that have LTS, which as of time of writing is only `v12` and `v14`. As a result, some dependencies that rely on `v15` or `v16` will fail to build using Vercel's pipeline.

![image](https://user-images.githubusercontent.com/1128312/148585887-17368203-48ab-4850-97d0-0c6b2f710097.png)

## (Optional) Possible Solution

We can downgrade our dependencies (see hoprnet/hoprnet#3244) but that might require a lot of work which might have unexpected consequences in the actual protocol.

## Steps to Reproduce

1. See #80

## Context (Environment)

For deploying MyneChat, we need a quick hosting provider, but since our application is a front-end application, we might not need to have a runtime (e.g. a node.js server) but instead a simple hosting provider. It would be easier to simply use another provider or pre-build the application before hosting it into Vercel.

**Comment by jjperezaguinaga** [comment]
Upon discussing with the team (@tolbrino @nionis), we'll pick the path of least resistance and introduce a middle step to allow deploying to Vercel but building in GH to avoid stumbling upon this issue for now. An alternative, to be considered later (if at all), will be to migrate to [Fleek](https://fleek.co/) or host the app in [Homescreen](https://docs.siasky.net/integrations/homescreen).

**Comment by jjperezaguinaga** [comment]
#82 didn't solve the issue and now we have #84 as a workaround.

**Comment by jjperezaguinaga** [comment]
Upon discussing with @nionis the downsides of migrating to Vercel, I explored a few alternatives to keep our codebase sane while avoiding switching providers due to incompatible runtimes.

The result is the following, which seems to be a sane compromise:

- The main [function](https://github.com/hoprnet/hoprnet/blob/059250384a04463fa1d1068dde38697ce683c817/packages/utils/src/libp2p/verifySignatureFromPeerId.ts#L15-L18) used in the app was copied verbatim from its implementation, by simply adding the minimal amount of dependencies (`peer-id`, `@ethersproject/bytes`) needed for it to work. This way we ensure we avoid bloating our project with everything `@hoprnet/hopr-utils`.
- `@hoprnet/hopr-utils` was _still_ added to the project, but instead of using the `latest` tag, we used the last known version to be compatible with `v14`. Thus, we can still install it and use it for testing (needed for running `yarn test` w/o having to rewrite all existing utilities) w/o forcing the project to be maintainable for external apps (i.e. this one). Further information can be seen in hoprnet/hoprnet#3244.

As of e06a202, the build is back to work and I can continue implementing the needed issues. The efforts done in #82 and #83 need to be removed (see #86), with learnings to be shared with the Vercel team in #84 (see a comment at the end). For now, it makes the most sense to retain MyneChat in Vercel, but other static-first providers (e.g. Fleek, GitHub Pages, Netlify, Surge, Cloudflare Pages) should not be dismissed.
